### PR TITLE
Add support for dynamic domain list in SecureLink

### PR DESCRIPTION
**Pull request by easonyq**
[AB#503998](https://dev.azure.com/ceapex/d3d54af3-265a-4f18-95f6-9a46397ca583/_workitems/edit/503998)

Changes:
1. Pull 'microsoft.domain' from Taxonomy Service
2. Pass it to Validation lib
3. Bump validation lib version to latest beta

**Please review but don't merge it. I'll merge it when everything is ready**

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/docfx?pullRequest=7757) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/docfx?pullRequest=7757) before we can accept your contribution.<br/><br/>:x: easonyq [sign now](https://cla.dotnetfoundation.org/dotnet/docfx?pullRequest=7757)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/docfx?pullRequest=7757) it.</sub>

File: docs/specs/validation/link-validation.yml [hunk 1]
```diff
@@ -482,6 +482,58 @@ outputs:
...
+        ]
+      }
+    }
+  allowlist.json: |
+    {
+        "microsoft.domain": {
+            "nestedValue": "",
+            "nestedTaxonomy": [
+                "microsoft.com",
+                "youtube.com"
```
Comment (herohua) [inline docs/specs/validation/link-validation.yml:520]: How about using another domain for testing? `youtube.com` would be a bit confusing as it is not a Microsoft domain.

File: docs/specs/validation/link-validation.yml [hunk 2]
```diff
@@ -482,6 +482,58 @@ outputs:
...
+      "links": {
+        "rules": [
+          {
+            "type": "SecureLinks",
+            "message": "Link '{0}' is insecure. Links to Microsoft sites must use 'https' instead of 'http'.",
+            "code": "insecure-link-2",
+            "severity": "INFO",
+            "contentTypes": ["conceptual"],
+            "useDynamicDomains": true,
+            "domains": [
```
Comment (herohua) [inline docs/specs/validation/link-validation.yml:501]: Should we add a case to show that we don't use this `domains` list here?

Comment (easonyq) [reply docs/specs/validation/link-validation.yml:501]: There is one. Look at the 4th line in file1.md, `http://www.google.com`. It is in fixed domains but not in dynamic domains, so it won't be reported.

File: docs/specs/validation/link-validation.yml (hunk 1 above)
Comment (easonyq) [reply docs/specs/validation/link-validation.yml:520]: Well, OK.

**Review by herohua — APPROVED** [review]

**Pull request merged**